File: 코드/ch8/clip4/boj1316/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n;
        n = sc.nextInt();
        int cnt = 0;
        for (int i = 0; i < n; i++) {
            boolean[] check = new boolean[26];
            boolean isGroupWord = true;
            char[] str = sc.next().toCharArray();
            for (int j = 0; j < str.length - 1; j++) {
                char now = str[j];
                char next = str[j + 1];
                if(now == next) continue;

                if (check[now - 'a']) {
                    isGroupWord = false;
                    break;
                }
                check[now - 'a'] = true;
            }
            if (check[str[str.length - 1] - 'a']) isGroupWord = false;
            if (isGroupWord) cnt++;

        }
        System.out.println(cnt);
    }
}

File: 코드/ch8/clip4/boj1316/main.py
import sys

n = int(sys.stdin.readline().rstrip())
cnt = 0

for _ in range(n):
    check = [False] * 26
    is_group_word = True
    word = sys.stdin.readline().rstrip()
    
    for j in range(len(word) - 1):
        now = word[j]
        next_ = word[j + 1]
        if now == next_:
            continue
        if check[ord(now) - ord('a')]:
            is_group_word = False
            break
        check[ord(now) - ord('a')] = True
        
    if check[ord(word[-1]) - ord('a')]:
        is_group_word = False
    if is_group_word:
        cnt += 1

sys.stdout.write(str(cnt) + '\n')
